src/resources/adc_esp32.cc

```diff
@@ -104,27 +106,26 @@ PRIMITIVE(init) {
     }
   }
 
-  ADCState* state = _new ADCState(unit, chan);
+  AdcState* state = _new AdcState(group);
```

Comment:
This code is OK, but it's actually written the wrong way around, before you even got to it.

The thing is that allocations off the GCed heap and on the GCed heap can fail, and the reaction to both is to return from the primitive with an error.  The primitive will be retried after a GC.

The state is allocated with new so it must be deleted with delete before we return an error.  The proxy is allocated on the GCed heap, so you don't need to free it, the GC will clean it up.  Therefore it would be better to allocate the proxy first, then create the AdcState.  That way you can drop the "delete state" because the state hasn't been 'new'ed yet when the proxy fails to allocate.

It's not a huge thing, it's just neater that way around.